tools/autograd/templates/python_torch_functions.cpp

```diff
@@ -121,7 +112,10 @@ static PyObject * THPVariable_arange(PyObject* self, PyObject* args, PyObject* k
     if (r.isNone(1)) {
       auto end = r.scalar(0);
       // NOTE: r.scalartype(X) gives the default dtype if r.isNone(X)
-      auto scalarType = r.isNone(2) && allIntegral({end}) ? at::ScalarType::Long : r.scalartype(2);
+      c10::optional<ScalarType> scalarType = c10::nullopt;
```

Comment:
It's worth exploring if we can delete `THPVariable_arange` from this file (templates/python_torch_functions.cpp) and see if we can auto-generate the arange code. From reading the original arange pull request, it looks like the reason why we special-cased it is to support the custom scalarType logic; now that we've pushed the logic to the arange operator itself we shouldn't this manual binding anymore.

Reply:
Okay, I looked into this a little. We can't actually autogenerate `THPVariable_arange` because the generated code uses `r.scalartype(X)`, which "gives the default dtype if r.isNone(X)" (thank you to whoever wrote that comment above this line). 

Given that, our two options are:
1) Fix the python arg parser
2) Accept this change.

I'm inclined to go with option (2) because this change does not make the code any worse than before.